### ISSUE TITLE
Can't connect to redis socket after upgrade to 4.18

### ISSUE THREAD

**Issue by steffan-BW** [pretix/pretix#3193]
### Problem and impact

When upgrading Pretix to 4.18, running as a docker container, from 4.17 to 4.18 and running `docker exec -it pretix.service pretix upgrade` the following error is thrown:

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/pretix/src/pretix/__main__.py", line 30, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/management/commands/migrate.py", line 75, in handle
    self.check(databases=[database])
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 419, in check
    all_issues = checks.run_checks(
                 ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/checks/registry.py", line 76, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/checks/urls.py", line 13, in check_url_config
    return check_resolver(resolver)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/checks/urls.py", line 23, in check_resolver
    return check_method()
           ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/urls/resolvers.py", line 416, in check
    for pattern in self.url_patterns:
                   ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
                                         ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/urls/resolvers.py", line 602, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
                       ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
                                         ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/urls/resolvers.py", line 595, in urlconf_module
    return import_module(self.urlconf_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/pretix/src/pretix/multidomain/maindomain_urlconf.py", line 46, in <module>
    from pretix.urls import common_patterns
  File "/pretix/src/pretix/urls.py", line 40, in <module>
    import pretix.control.urls
  File "/pretix/src/pretix/control/urls.py", line 40, in <module>
    from pretix.control.views import (
  File "/pretix/src/pretix/control/views/orders.py", line 120, in <module>
    from pretix.control.forms.orders import (
  File "/pretix/src/pretix/control/forms/orders.py", line 594, in <module>
    class OrderContactForm(forms.ModelForm):
  File "/pretix/src/pretix/control/forms/orders.py", line 601, in OrderContactForm
    class Meta:
  File "/pretix/src/pretix/control/forms/orders.py", line 605, in Meta
    'phone': WrappedPhoneNumberPrefixWidget(),
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/pretix/src/pretix/base/forms/questions.py", line 306, in __init__
    widgets = (WrappedPhonePrefixSelect(initial), forms.TextInput(attrs=attrs))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/pretix/src/pretix/base/forms/questions.py", line 279, in __init__
    choices += get_phone_prefixes_sorted_and_localized()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/pretix/src/pretix/helpers/countries.py", line 104, in get_phone_prefixes_sorted_and_localized
    val = cache.get(cache_key)
          ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django_redis/cache.py", line 91, in get
    value = self._get(key, default, version, client)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django_redis/cache.py", line 31, in _decorator
    return method(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django_redis/cache.py", line 98, in _get
    return self.client.get(key, default=default, version=version, client=client)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django_redis/client/default.py", line 258, in get
    value = client.get(key)
            ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/commands/core.py", line 1790, in get
    return self.execute_command("GET", name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/client.py", line 1255, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/connection.py", line 1441, in get_connection
    connection.connect()
  File "/usr/local/lib/python3.11/site-packages/redis/connection.py", line 698, in connect
    sock = self.retry.call_with_retry(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/retry.py", line 46, in call_with_retry
    return do()
           ^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/connection.py", line 699, in <lambda>
    lambda: self._connect(), lambda error: self.disconnect(error)
            ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/connection.py", line 1171, in _connect
    sock.settimeout(self.socket_timeout)
                    ^^^^^^^^^^^^^^^^^^^
AttributeError: 'UnixDomainSocketConnection' object has no attribute 'socket_timeout'
```

When downgrading back to 4.17 the container has no trouble connecting to the redis socket and runs normally.

### Steps to reproduce

1 Install Pretix 4.17 on an Ubuntu server following the https://docs.pretix.eu/en/latest/admin/installation/docker_smallscale.html guide.
2 Update to 4.18 and run `docker exec -it pretix.service pretix upgrade`

### Operating system, dependency versions

Ubuntu 22.04.2 LTS, Redis Server 6.0.16

### Version

4.18

**Comment by raphaelm** [comment]
This is really weird, Im curious if anyone else can reproduce this as we are not seeing it in production.

**Comment by stapelberg** [comment]
I ran into this too, yesterday, and can reproduce it.

The triggering condition seems to be configuring `pretix.cfg` to use UNIX sockets for connecting celery to redis:

```
[celery]
backend=redis+socket:///var/run/redis/redis.sock?virtual_host=1
broker=redis+socket:///var/run/redis/redis.sock?virtual_host=2
```

Then, the following command works for me (brings up an empty pretix):

```
docker run -v /tmp/etc/pretix:/etc/pretix -v /var/run/redis:/var/run/redis \
  -ti -p 8000:80 --env NUM_WORKERS=2 pretix/standalone:4.17
```

But running the same command with 4.18 (or :stable) fails:

<details>
<summary>docker run 4.18 log</summary>

```
docker run -v /tmp/etc/pretix:/etc/pretix -v /var/run/redis:/var/run/redis -ti -p 8000:80 --env NUM_WORKERS=2 pretix/standalone:4.18 
Unable to find image 'pretix/standalone:4.18' locally
4.18: Pulling from pretix/standalone
3e440a704568: Already exists 
68a71c865a2c: Already exists 
670730c27c2e: Already exists 
5a7a2c95f0f8: Already exists 
6d627e120214: Already exists 
f8c6dc678081: Already exists 
48bd2de548fc: Already exists 
e69bcee2d314: Already exists 
284a2f237609: Already exists 
b37519b79a70: Pull complete 
332b76c61633: Pull complete 
21e5e720a9ef: Pull complete 
b22f07a59114: Pull complete 
435dfcc70b32: Pull complete 
4c6133244879: Pull complete 
67c1b4e66133: Pull complete 
c8a8b39db185: Pull complete 
ef0f9b400882: Pull complete 
df75e505e994: Pull complete 
96490c6cb44e: Pull complete 
a72688093fa2: Pull complete 
8a54d218e707: Pull complete 
Digest: sha256:f1d988195adfe94383b0e5231321166d5d5bc053f8f6e0d0bb2d760c3130a297
Status: Downloaded newer image for pretix/standalone:4.18
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/pretix/src/pretix/__main__.py", line 30, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/management/commands/migrate.py", line 75, in handle
    self.check(databases=[database])
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 419, in check
    all_issues = checks.run_checks(
                 ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/checks/registry.py", line 76, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/checks/urls.py", line 13, in check_url_config
    return check_resolver(resolver)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/checks/urls.py", line 23, in check_resolver
    return check_method()
           ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/urls/resolvers.py", line 416, in check
    for pattern in self.url_patterns:
                   ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
                                         ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/urls/resolvers.py", line 602, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
                       ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
                                         ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/urls/resolvers.py", line 595, in urlconf_module
    return import_module(self.urlconf_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/pretix/src/pretix/multidomain/maindomain_urlconf.py", line 46, in <module>
    from pretix.urls import common_patterns
  File "/pretix/src/pretix/urls.py", line 40, in <module>
    import pretix.control.urls
  File "/pretix/src/pretix/control/urls.py", line 40, in <module>
    from pretix.control.views import (
  File "/pretix/src/pretix/control/views/orders.py", line 120, in <module>
    from pretix.control.forms.orders import (
  File "/pretix/src/pretix/control/forms/orders.py", line 594, in <module>
    class OrderContactForm(forms.ModelForm):
  File "/pretix/src/pretix/control/forms/orders.py", line 601, in OrderContactForm
    class Meta:
  File "/pretix/src/pretix/control/forms/orders.py", line 605, in Meta
    'phone': WrappedPhoneNumberPrefixWidget(),
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/pretix/src/pretix/base/forms/questions.py", line 306, in __init__
    widgets = (WrappedPhonePrefixSelect(initial), forms.TextInput(attrs=attrs))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/pretix/src/pretix/base/forms/questions.py", line 279, in __init__
    choices += get_phone_prefixes_sorted_and_localized()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/pretix/src/pretix/helpers/countries.py", line 104, in get_phone_prefixes_sorted_and_localized
    val = cache.get(cache_key)
          ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django_redis/cache.py", line 91, in get
    value = self._get(key, default, version, client)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django_redis/cache.py", line 31, in _decorator
    return method(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django_redis/cache.py", line 98, in _get
    return self.client.get(key, default=default, version=version, client=client)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django_redis/client/default.py", line 258, in get
    value = client.get(key)
            ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/commands/core.py", line 1790, in get
    return self.execute_command("GET", name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/client.py", line 1255, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/connection.py", line 1441, in get_connection
    connection.connect()
  File "/usr/local/lib/python3.11/site-packages/redis/connection.py", line 698, in connect
    sock = self.retry.call_with_retry(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/retry.py", line 46, in call_with_retry
    return do()
           ^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/connection.py", line 699, in <lambda>
    lambda: self._connect(), lambda error: self.disconnect(error)
            ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/redis/connection.py", line 1171, in _connect
    sock.settimeout(self.socket_timeout)
                    ^^^^^^^^^^^^^^^^^^^
AttributeError: 'UnixDomainSocketConnection' object has no attribute 'socket_timeout'
2023-03-29 15:17:48,665 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to run as root, you can set user=root in the config file to avoid this message.
2023-03-29 15:17:48,665 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to run as root, you can set user=root in the config file to avoid this message.
2023-03-29 15:17:48,665 INFO Included extra file "/etc/supervisord/base.conf" during parsing

```

</details>

**Comment by stapelberg** [comment]
I wonder if this is related to https://github.com/redis/redis-py/issues/2629 at all?

**Comment by raphaelm** [comment]
Yes, that seems related. We do depend on redis==4.5.* starting with 4.18. Can you check the docker tag ``4.18.2`` released today? Maybe it already has the fixed version it it

**Comment by stapelberg** [comment]
> Can you check the docker tag `4.18.2` released today? Maybe it already has the fixed version it it

Sure. Unfortunately, 4.18.2 also fails with the same error message, so it does not seem to include the fix yet.

**Comment by stapelberg** [comment]
Is there a workaround? I think going back to 4.16 left me with an already-migrated database, and now orders and adding products doesn’t work anymore. I’m not familiar with how version control works in Python, or how to do that within the pretix Dockerfile.

**Comment by mvanderkruk** [comment]
Same issue here. Checked with pretix/standalone:stable as well as pretix/standalone:4.18.2 docker image

**Comment by raphaelm** [comment]
Thanks, I'll likely do another bugfix release with a pinned version tomorrow!

**Comment by raphaelm** [comment]
Fixed in ``4.18.2.post1`` :)
